redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/termination/handler/DeregisterDatabaseServerHandler.java

```diff
@@ -21,21 +25,37 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(DeregisterDatabaseServerHandler.class);
 
+    private static final long DEFAULT_WORKSPACE = 0L;
+
     @Inject
     private EventBus eventBus;
 
+    @Inject
+    private DatabaseConfigRepository databaseConfigRepository;
+
+    @Inject
+    private DatabaseServerConfigService databaseServerConfigService;
+
+    @Inject
+    private DBStackService dbStackService;
+
     @Override
     public String selector() {
         return EventSelectorUtil.selector(DeregisterDatabaseServerRequest.class);
     }
 
     @Override
     public void accept(Event<DeregisterDatabaseServerRequest> event) {
-        RedbeamsEvent request = event.getData();
+        DeregisterDatabaseServerRequest request = event.getData();
         Selectable response = new DeregisterDatabaseServerSuccess(request.getResourceId());
 
-        // TODO: Actually deregister database server
-        LOGGER.info("A database server would be deregistered here.");
+        DBStack dbStack = request.getDbStack();
+
+        DatabaseServerConfig dbServerConfig =
+                databaseServerConfigService.getByName(DEFAULT_WORKSPACE, dbStack.getEnvironmentId(), dbStack.getName());
+
+        databaseServerConfigService.archive(dbServerConfig);
```

Comment:
I suggest wrapping these in a try-catch and sending a failure if they don't work. It could be done now, or we can pop it in later with the upcoming status updates.

Reply:
I'll pop this in now and re-push, then merge.